packages/mask/src/plugins/Pets/SNSAdaptor/PetSetDialog.tsx

```diff
@@ -0,0 +1,337 @@
+
+const useStyles = makeStyles()((theme) => ({
+    desBox: {
+        display: 'flex',
+        justifyContent: 'space-between',
+        marginTop: theme.spacing(3),
+    },
+    des: {
+        color: '#7b8192',
+        fontSize: '12px',
+    },
+    input: {
+        margin: theme.spacing(2, 0, 0),
+    },
+    inputOption: {
+        margin: theme.spacing(4, 0, 0),
+    },
+    inputBorder: {
+        borderRadius: theme.spacing(1),
+        padding: theme.spacing(1),
+    },
+    inputArea: {
+        borderRadius: theme.spacing(1),
+        padding: theme.spacing(2),
+    },
+    menuItem: {
+        width: '100%',
+    },
+    btn: {
+        margin: theme.spacing(8, 0, 2),
+    },
+    thumbnail: {
+        width: 25,
+        height: 25,
+        objectFit: 'cover',
+        margin: theme.spacing(0, 1, 0, 0),
+        display: 'inline-block',
+        borderRadius: 4,
+    },
+    glbIcon: {
+        width: 15,
+        height: 18,
+        marginLeft: theme.spacing(1),
+    },
+    itemFix: {
+        display: 'flex',
+        alignItems: 'center',
+        width: '100%',
+    },
+    itemTxt: {
+        flex: 1,
+        marginLeft: theme.spacing(0.5),
+    },
+    prevBox: {
+        margin: theme.spacing(2, 0, 0),
+        border: '1px dashed #ccc',
+        borderRadius: 4,
+        height: 'calc(100% - 16px)',
+        boxSizing: 'border-box',
+        padding: 4,
+    },
+    boxPaper: {
+        backgroundColor: theme.palette.mode === 'dark' ? '#1B1E38' : '#FFFFFF',
+        marginBottom: 10,
+        boxShadow: theme.palette.mode === 'dark' ? '0 0 5px #FFFFFF' : '0 0 5px #CCCCCC',
+    },
+}))
+
+interface PetSetDialogProps {
+    configNFTs: Record<string, Constant> | undefined
+    onClose: () => void
+}
+
+export function PetSetDialog({ configNFTs, onClose }: PetSetDialogProps) {
+    const { t } = useI18N()
+    const classes = useStylesExtends(useStyles(), {})
+    const { showSnackbar } = useCustomSnackbar()
+    const [loading, setLoading] = useState(false)
+    const checked = useValueRef<boolean>(petShowSettings)
+
+    const user = useUser()
+    const nfts = useNFTs(user, configNFTs)
+    const extraData = useNFTsExtra(configNFTs)
+    const [collection, setCollection] = useState<FilterContract>(initCollection)
+    const [isCollectionsError, setIsCollectionsError] = useState(false)
+
+    const [metaData, setMetaData] = useState<PetMetaDB>(initMeta)
+    const [isImageError, setIsImageError] = useState(false)
+    const [isTipShow, setIsTipShow] = useState(false)
+    const [holderChange, setHolderChange] = useState(true)
+    const [tokenInfoSelect, setTokenInfoSelect] = useState<OwnerERC721TokenInfo | null>(null)
+    const [inputTokenName, setInputTokenName] = useState('')
+
+    useEffect(() => {
+        setMetaData(initMeta)
+        setCollection(initCollection)
+        setTokenInfoSelect(null)
+        setInputTokenName('')
+    }, [])
+
+    let timer: NodeJS.Timeout | null = null
```

Comment:
No. This isn't correct. You must store it in a `useRef` otherwise it won't work if the component gets rerendered.